### PR TITLE
Standardise documentation structure and correct broken links

### PR DESCRIPTION
**Pull request by TimothyJones**
When changing the documentation, I found the structure difficult to follow. This PR ensures that the folder structure matches the structure in the table of contents, hopefully making it easier for future contributors.

I've also renamed some of the files whose filenames (and therefore URLs) no longer reflected the content.

As part of this process, I've been through all of the internal and external links to confirm that they work. This PR also corrects a few broken/outdated external links.

Accepting this PR may have the side effect that some direct links into the documentation may break. Personally I think that having better URLs and nicer structured documentation is worth this cost, but I'll leave that up to the maintainers to discuss.

### CODE REVIEW

File: getting_help/README.md [hunk 1]
```diff
@@ -8,6 +8,5 @@ Join us in Slack:
 or, you can get help on Pact from the following channels:
 
 * **Stack Overflow**: https://stackoverflow.com/questions/tagged/pact
-* **Twitter:** (@pact-up)[https://twitter.com/pact_up)
+* **Twitter:** [@pact-up](https://twitter.com/pact_up)
```
Comment (mboudreau) [inline getting_help/README.md]: it's `pact_up`

File: contributing/README.md [hunk 2]
```diff
@@ -10,6 +10,6 @@ simply do the following:
 5. Create new Pull Request
 
 If you would like to see a bigger feature, or see `Pact` implemented in another
-language, please check out the [Pact specification](https://github.com/bethesque/pact-specification) and have a chat to one of us on the [pact-dev Google group](https://groups.google.com/forum/#!forum/pact-dev).
+language, please check out the [Pact specification](https://github.com/pact-foundation/pact-specification) and have a chat to one of us on the [pact-dev Google group](https://groups.google.com/forum/#!forum/pact-dev).
```
Comment (mboudreau) [inline contributing/README.md]: We should probably remove the pact-dev google group and specify our slack at http://slack.pact.io instead

File: getting_help/README.md [hunk 3]
```diff
@@ -8,6 +8,5 @@ Join us in Slack:
...
 * **Gitter:** Join the chat at https://gitter.im/realestate-com-au/pact and https://gitter.im/DiUS/pact-jvm (_NOTE_: we are gradully moving this to Slack)
```
Comment (mboudreau) [inline getting_help/README.md]: Remove gitter, replace with Slack.

**Comment by mboudreau** [comment]
@TimothyJones slack script?

File: getting_help/README.md [hunk 4]
```diff
@@ -1,12 +1,7 @@
 # Getting help
 
-Join us in Slack:
-
-<script async defer src="https://pact-slack.herokuapp.com/slackin.js"></script>
```
Comment (TimothyJones) [inline getting_help/README.md]: This is the slack script I removed.

**Comment by mboudreau** [comment]
Ah right, the old slackin.  I think that heroku account got removed.

File: getting_help/README.md (hunk 4 above)
Comment (mefellows) [reply getting_help/README.md]: could replace it with the SVG (the heroku link should still work, but perhaps gitbook doesn't like the `<script>` element:

```
[![slack](http://slack.pact.io/badge.svg)](http://slack.pact.io)
```
or
```
<a href="http://slack.pact.io"><img src="http://slack.pact.io/badge.svg"></a>
```

Comment (TimothyJones) [reply getting_help/README.md]: Aha! Thanks. I've put that in.

**Comment by mboudreau** [comment]
shoot, sorry :P

**Comment by mboudreau** [comment]
@TimothyJones you are now part of the collaborators team :)  Please continue to do PRs to keep people informed and the great work.

**Pull request merged**